src/Sentry.OpenTelemetry/SentryPropagator.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.AspNetCore.Http;
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+
+namespace Sentry.OpenTelemetry;
+
+/// <summary>
+/// Sentry OpenTelemetry Propagator.
+/// Injects and extracts both <c>sentry-trace</c> and <c>baggage</c> headers from carriers.
+/// </summary>
+public class SentryPropagator : BaggagePropagator
+{
+    /// <inheritdoc />
+    public override ISet<string> Fields => new HashSet<string>
+    {
+        SentryTraceHeader.HttpHeaderName,
+        BaggageHeader.HttpHeaderName
+    };
+
+    private static class OTelKeys
+    {
+        public const string SentryBaggageKey = "sentry.baggage";
+        public const string SentryTraceKey = "sentry.trace";
+    }
+
+    /// <inheritdoc />
+    public override PropagationContext Extract<T>(PropagationContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+    {
+        var result = base.Extract(context, carrier, getter);
+
+        if (carrier is not HttpRequest request)
+        {
+            return result;
+        }
+
+        var modifiedContext = result.Baggage;
+
+        if (TryGetSentryTraceHeader(request.HttpContext) is not {} sentryTraceHeader)
+        {
+            return result;
+        }
+        modifiedContext.SetBaggage(OTelKeys.SentryTraceKey, sentryTraceHeader.ToString());
+
+        // Sentry uses the word Baggage (https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#baggage)
+        // to mean something a bit different to OTEL (https://opentelemetry.io/docs/concepts/signals/baggage/)... so
+        // the names here are a bit confusing.
+        var baggageHeader = TryGetBaggageHeader(request.HttpContext)
+                            ?? BaggageHeader.Create(new List<KeyValuePair<string, string>>());
+        modifiedContext.SetBaggage(OTelKeys.SentryBaggageKey, baggageHeader.ToString());
+
+        var otelSpanContext = new ActivityContext(
+            sentryTraceHeader.TraceId.AsActivityTraceId(),
+            sentryTraceHeader.SpanId.AsActivitySpanId(),
+            sentryTraceHeader.IsSampled is true ? ActivityTraceFlags.Recorded : ActivityTraceFlags.None,
+            null,
+            true
+            );
+
+        // TODO: Understand why this is needed in the Java implementation
```

Comment:
@bruno-garcia do you have any idea why they've done [this in the Java implementation](https://github.com/getsentry/sentry-java/blob/cb2fb895cc8d31a4be7025bce2ea536fb7d4d055/sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentryPropagator.java#L81)?

From [the source code for otel's java sdk](https://github.com/open-telemetry/opentelemetry-java/blob/bcec7e9380c943be3c3764efa22fbb10ddd6473f/api/all/src/main/java/io/opentelemetry/api/trace/Span.java#L79):
```
  /**
   * Returns a non-recording {@link Span} that holds the provided {@link SpanContext} but has no
   * functionality. It will not be exported and all tracing operations are no-op, but it can be used
   * to propagate a valid {@link SpanContext} downstream.
   */
  static Span wrap(SpanContext spanContext) {
    if (spanContext == null) {
      ApiUsageLogger.log("context is null");
      return getInvalid();
    }
    if (!spanContext.isValid()) {
      return getInvalid();
    }
    return PropagatedSpan.create(spanContext);
  }
```

The only reason I can think why they're doing this is to get a Span.noOp... and they appear to have littered the [rest of their codebase](https://github.com/getsentry/sentry-java/pull/2361/files) with checks like `span != null && !span.isNoOp ` instead of `span != null` to determine whether tracing gets performed... so I think they're using this as a way to disable Sentry's built in tracing mechanism when Otel tracing is being used, but wanted to double check I'd understood correctly.

Reply:
> so I think they're using this as a way to disable Sentry's built in tracing mechanism when Otel tracing is being used, but wanted to double check I'd understood correctly.

That's right, my understanding is that either one or the other gets active. cc @AbhiPrasad to confirm

Reply:
Yup we either choose one or the other. See https://develop.sentry.dev/sdk/performance/opentelemetry/#step-6-add-instrumenter-option-to-sentry-sdk for more details.